README.md
# flat_segmented_control

Segemented Control with TabView for Flutter. Inspired by SegmentedControl (iOS) and Material Design.

![Gif Example](https://raw.githubusercontent.com/dhimasdewanto/flat_segmented_control/master/readme_assets/readme.gif)

## How to use

1. Add flat_segmented_control to your package's pubspec.yaml, then intall it.

2. Import package.

```
import 'package:flat_segmented_control/flat_segmented_control.dart';
```

3. Add this to your child Widget.
```
FlatSegmentedControl(
    tabChildren: <Widget>[
        Container(
        height: 50.0,
        child: Center(child: Text("Tab 1")),
        ),
        Container(
        height: 50.0,
        child: Center(child: Text("Tab 2")),
        ),
        Container(
        height: 50.0,
        child: Center(child: Text("Tab 3")),
        ),
    ],
    childrenHeight: 150.0,
    children: <Widget>[
        Center(child: Text("View 1")),
        Center(child: Text("View 2")),
        Center(child: Text("View 3")),
    ],
),
```

## Check

1. Make sure length of children same as navChildren.
2. Change childrenHeight or childrenWidth if you desire.

CHANGELOG.md
## [0.0.1] - 2019-07-20

* Initial Release.

## [0.0.2] - 2019-07-20

* Now required to insert height of children.

## [0.1.0] - 2019-08-26

* Now based on TabBar and TabBarView. Constructor has also been changed. Read README.md for more info.

test/flat_segmented_control_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:flat_segmented_control/flat_segmented_control.dart';

void main() {
  
}


lib/flat_segmented_control.dart
library flat_segmented_control;

import 'package:flutter/material.dart';

class FlatSegmentedControl extends StatefulWidget {
  final List<Widget> children;

  final List<Widget> tabChildren;

  /// The color of the line that appears below the selected tab.
  ///
  /// Defaults to Theme's indicatorColor [Theme.of(context).indicatorColor].
  final Color indicatorColor;

  /// The color of selected tab labels.
  ///
  /// Defaults to text body2 color [Theme.of(context).textTheme.body2.color].
  final Color labelColor;

  /// The color of unselected tab labels.
  ///
  /// Defaults to text body2 color [Theme.of(context).textTheme.body2.color].
  final Color unselectedLabelColor;

  /// Whether the children can be swiped horizontally.
  ///
  /// Defaults to [false].
  final bool isChildrenSwipeable;

  /// Padding of tab menu, not tab children.
  ///
  /// Defaults to [EdgeInsets.zero]
  final EdgeInsetsGeometry tabPadding;

  /// The view padding.
  ///
  /// Defaults to [EdgeInsets.zero]
  final EdgeInsetsGeometry childrenPadding;

  /// Define the children width. Please change as desired.
  ///
  /// Defaults to device width.
  final double childrenWidth;

  /// Define the children height. Please change as desired.
  ///
  /// Defaults to device height.
  final double childrenHeight;

  FlatSegmentedControl({
    Key key,
    @required this.children,
    @required this.tabChildren,
    this.indicatorColor,
    this.labelColor,
    this.isChildrenSwipeable = false,
    this.unselectedLabelColor,
    this.tabPadding,
    this.childrenPadding,
    this.childrenWidth,
    this.childrenHeight,
  })  : assert(children != null),
        assert(tabChildren != null),
        assert(tabChildren.length == children.length),
        super(key: key);

  @override
  _FlatSegmentedControlState createState() => _FlatSegmentedControlState();
}

class _FlatSegmentedControlState extends State<FlatSegmentedControl>
    with SingleTickerProviderStateMixin {
  TabController _tabController;

  @override
  void initState() {
    super.initState();
    _tabController = TabController(
      vsync: this,
      length: widget.children.length,
    );
  }

  @override
  void dispose() {
    super.dispose();
    _tabController?.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        Container(
          padding: widget.tabPadding ?? EdgeInsets.zero,
          child: TabBar(
            controller: _tabController,
            tabs: widget.tabChildren,
            indicatorColor: widget.indicatorColor,
            labelColor:
                widget.labelColor ?? Theme.of(context).textTheme.body2.color,
            unselectedLabelColor: widget.unselectedLabelColor ??
                Theme.of(context).textTheme.body2.color,
          ),
        ),
        Container(
          height: widget.childrenHeight ?? MediaQuery.of(context).size.height,
          width: widget.childrenWidth ?? MediaQuery.of(context).size.width,
          padding: widget.childrenPadding ?? EdgeInsets.zero,
          child: TabBarView(
            controller: _tabController,
            physics: widget.isChildrenSwipeable
                ? ScrollPhysics()
                : NeverScrollableScrollPhysics(),
            children: widget.children,
          ),
        ),
      ],
    );
  }
}


example/lib/main.dart
import 'package:flat_segmented_control/flat_segmented_control.dart';
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Segmented Control Demo',
      home: MyHomePage(title: 'Segmented Control Demo'),
    );
  }
}

class MyHomePage extends StatelessWidget {
  final String title;

  MyHomePage({Key key, this.title}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(title),
      ),
      body: ListView(
        children: <Widget>[
          Center(
            child: FlutterLogo(
              size: 150.0,
            ),
          ),
          SizedBox(height: 20.0),
          FlatSegmentedControl(
            tabChildren: <Widget>[
              Container(
                height: 50.0,
                child: Center(child: Text("Tab 1")),
              ),
              Container(
                height: 50.0,
                child: Center(child: Text("Tab 2")),
              ),
              Container(
                height: 50.0,
                child: Center(child: Text("Tab 3")),
              ),
            ],
            childrenHeight: 150.0,
            children: <Widget>[
              Center(child: Text("View 1")),
              Center(child: Text("View 2")),
              Center(child: Text("View 3")),
            ],
          ),
          SizedBox(height: 20.0),
          Text(
            "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
          ),
        ],
      ),
    );
  }
}
